terraform/hcl2json_test.go

```diff
@@ -939,6 +939,182 @@ variable "dummy" {
 				"debugLogs": map[string]interface{}{},
 			},
 		},
+		{
+			name: "Correctly dereferencing local values in a single file",
```

Comment:
These tests are great, but I would still add a test case for having duplicates locals as this was discussed twice and it's not as easy to spot. It doesn't have to be in this PR, and it can be enhanced in a different one.

Tests are also documentation of the expected behaviour so that the next person does not have to go through the entire implementation to spot something like this.

Reply:
> test case for having duplicates locals

`terraform validate` will fail in this case, why do we need to test such cases?

Reply:
I'm going to merge the PR so let's continue the convo in [this thread](https://snyk.slack.com/archives/C030WL9BLSW/p1646575964229079).